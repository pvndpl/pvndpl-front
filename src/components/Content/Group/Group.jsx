
import styles from './Group.module.css';
import HeaderContent from "../HeaderContent/HeaderContent";
import axios from '../../../redux/axios';
import Cookies from 'js-cookie';
import React, { useState, useEffect } from 'react';
import { Link } from "react-router-dom";


const Aboba = (props) => {
    const { username, name, id } = props.data;
    return (
        <a href={`http://localhost:3000/pvndpl-front/users/${id}`}>
            <div className={styles.users}>
                <div className={styles.user}>
                    <div className={styles.preview}>
                        <img className={styles.headerInfoImg}
                            src={"https://damion.club/uploads/posts/2022-01/thumbs/1642968512_17-damion-club-p-tupaya-obezyana-19.jpg"} />
                        <p className={styles.name}>{name}</p>
                        <p className={styles.tag}>@{username}</p>
                    </div>
                </div>
            </div>
        </a>
    )
}


const Group = () => {    const [conversations, setConversations] = useState([]);

    useEffect(() => {
        getConversations()
        getConversations1()
        user()
    }, [])

    const [username1, setUsername1] = useState();
    const [username, setUsername] = useState();

    const user = (e) => {
        axios.get(
            `/user-info`,
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            setUsername1(response)
        });
    }

    const getConversations = (e) => {
        axios.get(
            '/profile',
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            setUsername(response)
        });
    }

    const getConversations1 = () => {
        axios.get(
            `/subscribers`,
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            let newConversations = response.data.map(result => {
                return {
                    name: `${result.firstName} ${result.secondName}`,
                    id: result.id,
                    username: result.username
                };
            });
            setConversations([...conversations, ...newConversations])
        });
    }
    if (Array.isArray(conversations) && !!username1 && !!username) {
        const name1 = `${username1.data.firstname} ${username1.data.lastname}`;
        const count = username1.data.subscribersCount;
        return (<div className={styles.Friend} data-testid="Friend">
            <HeaderContent name={name1} section={"Подписчики"} count={count} />
            {
                conversations.map(conversation =>
                    <Aboba
                        data={conversation}
                    />
                )
            }
        </div>)
    }
}


export default Group;
