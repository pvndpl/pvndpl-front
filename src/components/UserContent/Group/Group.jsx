import axios from '../../../redux/axios';
import Cookies from 'js-cookie';
import React, { useState, useEffect } from 'react';
import { Link } from "react-router-dom";
import styles from './Group.module.css';
import HeaderContent from "../HeaderContent/HeaderContent";

const Aboba = (props) => {
    const { username, name, id } = props.data;
    return (
        <a href={`http://localhost:3000/pvndpl-front/users/${id}`}>
            <div className={styles.users}>
                <div className={styles.user}>
                    <div className={styles.preview}>
                        <img className={styles.headerInfoImg}
                            src={"https://sciencequizing.com/wp-content/uploads/2020/03/ccelebritiesfoto0_Portrait-of-a-monkey.jpg"} />
                        <p className={styles.name}>{name}</p>
                        <p className={styles.tag}>@{username}</p>
                    </div>
                </div>
            </div>
        </a>
    )
}

const Group = (props) => {
    const [conversations, setConversations] = useState([]);

    useEffect(() => {
        getConversations()
        getConversations1()
        user()
    }, [])

    const [username1, setUsername1] = useState();
    const [username, setUsername] = useState();

    const user = (e) => {
        axios.get(
            `/user-info/${props.path}`,
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            setUsername1(response)
        });
    }

    const getConversations = (e) => {
        axios.get(
            '/profile',
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            setUsername(response)
        });
    }

    const getConversations1 = () => {
        axios.get(
            `/subscriptions/${props.path}`,
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            let newConversations = response.data.map(result => {
                return {
                    name: `${result.firstName} ${result.secondName}`,
                    id: result.id,
                    username: result.username
                };
            });
            setConversations([...conversations, ...newConversations])
        });
    }
    if (Array.isArray(conversations) && !!username1 && !!username) {
        const name1 = `${username1.data.firstname} ${username1.data.lastname}`;
        const count = username1.data.subscribersCount;
        return (<div className={styles.Friend} data-testid="Friend">
            <HeaderContent name={name1} section={"Подписчики"} count={count} />
            {
                conversations.map(conversation =>
                    <Aboba
                        data={conversation}
                    />
                )
            }
        </div>)
    }
}


export default Group;
