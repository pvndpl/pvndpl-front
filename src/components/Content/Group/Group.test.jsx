import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Group from './Group';

describe('<Navbar />', () => {
  test('it should mount', () => {
    const element = render(<Group />);

    expect(element).toMatchSnapshot();

  });
});