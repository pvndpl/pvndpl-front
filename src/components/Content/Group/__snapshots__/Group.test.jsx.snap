// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> it should mount 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="Group"
      >
        <div
          class="HeaderContent"
          data-testid="HeaderContent"
        >
          <p
            class="name"
          >
            Александр Великодный
          </p>
          <h2
            class="section"
          >
            Группы
            <span
              class="count"
            >
              4
            </span>
          </h2>
        </div>
        <div
          class="groups"
        >
          <div
            class="group"
          >
            <div
              class="preview"
            >
              <img
                class="headerInfoImg"
                src="https://sun9-20.userapi.com/impg/Fu54-MoiNEDqdJ2KX9Hmac2BmLmW4Q693TllxQ/P25B3yjcQcs.jpg?size=1080x1080&quality=95&sign=dc5622e269dc500621a99bd5b3abf9fa&type=album"
              />
              <h2>
                 ПАТРИОТЫ | РОССИЯ !!!
              </h2>
            </div>
            <div
              class="info"
            >
              <div
                class="stat"
              >
                <p
                  class="text"
                >
                  2000000000
                </p>
                <p
                  class="title"
                >
                   Подписчиков
                </p>
              </div>
            </div>
          </div>
          <div
            class="group"
          >
            <div
              class="preview"
            >
              <img
                class="headerInfoImg"
                src="https://sun9-11.userapi.com/c108/g3156562/d_adfaa285.jpg"
              />
              <h2>
                 Дмитрий Goblin Пучков
              </h2>
            </div>
            <div
              class="info"
            >
              <div
                class="stat"
              >
                <p
                  class="text"
                >
                  200
                </p>
                <p
                  class="title"
                >
                   Подписчиков
                </p>
              </div>
            </div>
          </div>
          <div
            class="group"
          >
            <div
              class="preview"
            >
              <img
                class="headerInfoImg"
                src="https://sun9-24.userapi.com/s/v1/ig2/Pgn64dBG4BD5pqg0-x20PSc7cb8gabTEJAakVIQZyZ9MBhU2Q24phSvTNAgMDQE80_JY-Mbs5e3uEn0H4kwCz1VZ.jpg?size=100x100&quality=95&crop=79,66,393,393&ava=1"
              />
              <h2>
                 Минобороны России
              </h2>
            </div>
            <div
              class="info"
            >
              <div
                class="stat"
              >
                <p
                  class="text"
                >
                  20000000000
                </p>
                <p
                  class="title"
                >
                   Подписчиков
                </p>
              </div>
            </div>
          </div>
          <div
            class="group"
          >
            <div
              class="preview"
            >
              <img
                class="headerInfoImg"
                src="https://sun9-41.userapi.com/s/v1/if1/FM7AKIs9NAMRoDf1UBNl0uAIaubxKV1iln-UpAbpK7nR8EvR7be3wd7S5sLbFQXAddj55u28.jpg?size=100x100&quality=96&crop=457,81,513,513&ava=1"
              />
              <h2>
                Стрелков Игорь Иванович
              </h2>
            </div>
            <div
              class="info"
            >
              <div
                class="stat"
              >
                <p
                  class="text"
                >
                  200000
                </p>
                <p
                  class="title"
                >
                   Подписчиков
                </p>
              </div>
            </div>
          </div>
          <div
            class="group"
          >
            <div
              class="preview"
            >
              <img
                class="headerInfoImg"
                src="https://sun9-83.userapi.com/s/v1/ig2/HcH9F3oLxEPQhhfZMvBN8wW4ZPh-Zn4bYAfe5KWafDyi_5UO_KtOziyG27xBqqlHBdLr7hw6ZRJY77kFjpP1oDcz.jpg?size=100x100&quality=95&crop=32,42,577,577&ava=1"
              />
              <h2>
                Джанки
              </h2>
            </div>
            <div
              class="info"
            >
              <div
                class="stat"
              >
                <p
                  class="text"
                >
                  1
                </p>
                <p
                  class="title"
                >
                   Подписчиков
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="Group"
    >
      <div
        class="HeaderContent"
        data-testid="HeaderContent"
      >
        <p
          class="name"
        >
          Александр Великодный
        </p>
        <h2
          class="section"
        >
          Группы
          <span
            class="count"
          >
            4
          </span>
        </h2>
      </div>
      <div
        class="groups"
      >
        <div
          class="group"
        >
          <div
            class="preview"
          >
            <img
              class="headerInfoImg"
              src="https://sun9-20.userapi.com/impg/Fu54-MoiNEDqdJ2KX9Hmac2BmLmW4Q693TllxQ/P25B3yjcQcs.jpg?size=1080x1080&quality=95&sign=dc5622e269dc500621a99bd5b3abf9fa&type=album"
            />
            <h2>
               ПАТРИОТЫ | РОССИЯ !!!
            </h2>
          </div>
          <div
            class="info"
          >
            <div
              class="stat"
            >
              <p
                class="text"
              >
                2000000000
              </p>
              <p
                class="title"
              >
                 Подписчиков
              </p>
            </div>
          </div>
        </div>
        <div
          class="group"
        >
          <div
            class="preview"
          >
            <img
              class="headerInfoImg"
              src="https://sun9-11.userapi.com/c108/g3156562/d_adfaa285.jpg"
            />
            <h2>
               Дмитрий Goblin Пучков
            </h2>
          </div>
          <div
            class="info"
          >
            <div
              class="stat"
            >
              <p
                class="text"
              >
                200
              </p>
              <p
                class="title"
              >
                 Подписчиков
              </p>
            </div>
          </div>
        </div>
        <div
          class="group"
        >
          <div
            class="preview"
          >
            <img
              class="headerInfoImg"
              src="https://sun9-24.userapi.com/s/v1/ig2/Pgn64dBG4BD5pqg0-x20PSc7cb8gabTEJAakVIQZyZ9MBhU2Q24phSvTNAgMDQE80_JY-Mbs5e3uEn0H4kwCz1VZ.jpg?size=100x100&quality=95&crop=79,66,393,393&ava=1"
            />
            <h2>
               Минобороны России
            </h2>
          </div>
          <div
            class="info"
          >
            <div
              class="stat"
            >
              <p
                class="text"
              >
                20000000000
              </p>
              <p
                class="title"
              >
                 Подписчиков
              </p>
            </div>
          </div>
        </div>
        <div
          class="group"
        >
          <div
            class="preview"
          >
            <img
              class="headerInfoImg"
              src="https://sun9-41.userapi.com/s/v1/if1/FM7AKIs9NAMRoDf1UBNl0uAIaubxKV1iln-UpAbpK7nR8EvR7be3wd7S5sLbFQXAddj55u28.jpg?size=100x100&quality=96&crop=457,81,513,513&ava=1"
            />
            <h2>
              Стрелков Игорь Иванович
            </h2>
          </div>
          <div
            class="info"
          >
            <div
              class="stat"
            >
              <p
                class="text"
              >
                200000
              </p>
              <p
                class="title"
              >
                 Подписчиков
              </p>
            </div>
          </div>
        </div>
        <div
          class="group"
        >
          <div
            class="preview"
          >
            <img
              class="headerInfoImg"
              src="https://sun9-83.userapi.com/s/v1/ig2/HcH9F3oLxEPQhhfZMvBN8wW4ZPh-Zn4bYAfe5KWafDyi_5UO_KtOziyG27xBqqlHBdLr7hw6ZRJY77kFjpP1oDcz.jpg?size=100x100&quality=95&crop=32,42,577,577&ava=1"
            />
            <h2>
              Джанки
            </h2>
          </div>
          <div
            class="info"
          >
            <div
              class="stat"
            >
              <p
                class="text"
              >
                1
              </p>
              <p
                class="title"
              >
                 Подписчиков
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
